lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -1241,34 +1248,32 @@ selectCoinsForJoin ctx knownPools getPoolStatus pid wid = do
         (action, deposit) <- liftHandler
             $ W.joinStakePool @_ @s @k @n wrk curEpoch pools pid poolStatus wid
 
-        (wdrl, _mkRwdAcct) <- mkRewardAccountBuilder @_ @s @k @n ctx wid Nothing
         let txCtx = defaultTransactionCtx
-                { txWithdrawal = wdrl
-                , txDelegationAction = Just action
+                { txDelegationAction = Just action
```

Comment:
I've removed the call to `mkRewardAccountBuilder [...] Nothing` here which is actually a no-op. We don't allow withdrawals in delegation-like requests _yet_.